### PR TITLE
Plot 1d data without errorbars

### LINKED ISSUE #867: Plot 1D data without error bars
Currently this is not possible.
Syntax would be `plot(some1ddata, variances=False)`, in line with https://scipp.readthedocs.io/en/latest/visualization/plotting-in-depth.html#2-D-data-with-variances

Then, in `plot_1d.py`, one has to change all the instances of `if var.variances is not None:` to accommodate the value of `self.params["variances"]["show"]`, although right now this will only have a value if the first variable in the input has variances (https://github.com/scipp/scipp/blob/master/python/src/scipp/plot/slicer.py#L61). If input holds multiple variables, something smarter needs to be added.

### PR DESCRIPTION
**Pull request by nvaytet**
This PR makes it possible to hide/show error bars in 1D plots.

Example:

```Py
import numpy as np
import scipp as sc
from scipp import Dim
from scipp.plot import plot

d = sc.Dataset()
N = 50
d.coords[Dim.Tof] = sc.Variable([Dim.Tof], values=np.arange(N).astype(np.float64),
                                unit=sc.units.us)
d['Sample'] = sc.Variable([Dim.Tof], values=10.0*np.random.rand(N),
                          variances=np.square(np.random.rand(N)),
                          unit=sc.units.counts)
d['Background'] = sc.Variable([Dim.Tof], values=5.0*np.random.rand(N),
                              unit=sc.units.counts)
plot(d, variances=False)
plot(d, variances={"Sample": False, "Background": True})
```
Note that since `"Background"` does not have any variances, they will not be plotted by the last line (this prevents the plotting from failing altogether).

Fixes #867

### CODE REVIEW

File: python/src/scipp/plot/plot_1d.py [hunk 1]
```diff
@@ -99,6 +98,35 @@ def __init__(self,
...
 
+        # Determine whether error bars should be plotted or not
+        self.variances = {}
+        for name, var in self.scipp_obj_dict.items():
+            self.variances[name] = var.variances is not None
+        if variances is not None:
+            if isinstance(variances, bool):
+                for name, var in self.scipp_obj_dict.items():
+                    self.variances[
+                        name] = variances and var.variances is not None
```
Comment (SimonHeybrock) [inline python/src/scipp/plot/plot_1d.py]: Can just iterate the dict and do `&= variances` on each value?

File: python/src/scipp/plot/plot_1d.py [hunk 2]
```diff
@@ -99,6 +98,35 @@ def __init__(self,
...
+                    else:
+                        print("Warning: key {} was not found in list of "
+                              "entries to plot and will be ignored.".format(
+                                  name))
+            elif isinstance(variances, list):
+                keys = list(sorted(self.scipp_obj_dict.keys()))
+                for i, v in enumerate(variances):
+                    self.variances[
+                        keys[i]] = variances[i] and self.scipp_obj_dict[
+                            keys[i]].variances is not None
```
Comment (SimonHeybrock) [inline python/src/scipp/plot/plot_1d.py]: So this just accepts a list of `[True, False, ...]`? IS this really so useful, to justify all the extra code?

Comment (nvaytet) [reply python/src/scipp/plot/plot_1d.py]: I guess it's not very useful and actually error-prone. I'm supporting just a list of colors and markers, as I've often found this to be useful when plotting many curves and not really needing full control over the colors (as one would do in a waterfall plot), but I agree that for the variances you need to be more explicit.

File: python/src/scipp/plot/plot_1d.py (hunk 1 above)
Comment (nvaytet) [reply python/src/scipp/plot/plot_1d.py]: I don't think I can just do that because I also need to check that `var.variances` exists (i.e. is not `None`)

Comment (SimonHeybrock) [reply python/src/scipp/plot/plot_1d.py]: Didn't you do that already 5 lines above?

Comment (nvaytet) [reply python/src/scipp/plot/plot_1d.py]: :man_facepalming:

**Pull request merged**